cli/config.py

```diff
@@ -107,10 +108,17 @@ def load_config_from_api(auth_token, region, insecure):
     dsks = {dsk["station_sn"]: dsk for dsk in appapi.equipment_get_dsk_keys(station_sns=sns)["dsk_keys"]}
 
     # populate config object with printer list
+    # Sort the list of printers by printer.id
+    printers.sort(key=lambda p: p["station_id"])
     for pr in printers:
         station_sn = pr["station_sn"]
         config.printers.append(Printer(
+            id=pr["station_id"],
             sn=station_sn,
+            name=pr["station_name"],
+            model=pr["station_model"],
+            create_time=pr["create_time"],
```

Comment:
There's a subtle bug here, which doesn't manifest in our current code.

The config objects need to be structured data (so, `datetime()` instance instead of timestamp).

Here, we are creating a `Printer` object, where `.create_time` and `.update_time` are timestamps, not `datetime` objects.

So the return value from `load_config_from_api()` isn't a valid config, but because we happen to save and then load it, the timestamp is parsed when loading. So we mask that error. But if we had implemented, say, caching, we could have hit some very strange errors now :smile: 

Long story short, please apply this patch:


```diff
diff --git cli/config.py cli/config.py
index bd93525..0b0e116 100644
--- cli/config.py
+++ cli/config.py
@@ -117,8 +117,8 @@ def load_config_from_api(auth_token, region, insecure):
             sn=station_sn,
             name=pr["station_name"],
             model=pr["station_model"],
-            create_time=pr["create_time"],
-            update_time=pr["update_time"],
+            create_time=datetime.fromtimestamp(pr["create_time"]),
+            update_time=datetime.fromtimestamp(pr["update_time"]),
             mqtt_key=unhex(pr["secret_key"]),
             wifi_mac=pr["wifi_mac"],
             ip_addr=pr["ip_addr"],
diff --git cli/model.py cli/model.py
index 0cdaea7..6e9631c 100644
--- cli/model.py
+++ cli/model.py
@@ -23,6 +23,8 @@ def to_dict(self):
             res[k] = getattr(self, k)
             if v.type == bytes:
                 res[k] = enhex(res[k])
+            elif v.type == datetime:
+                res[k] = res[k].timestamp()
         return res

     @classmethod
```

Reply:
✅